tempodb/encoding/iterator_multiblock.go

```diff
@@ -108,32 +128,70 @@ func (i *multiblockIterator) iterate(ctx context.Context) {
 			}
 
 			comparison := bytes.Compare(currentID, lowestID)
+			foundLowerID := comparison == -1
+			shouldMerge := comparison == 0
+			// There is potential for an early dropout if comparison > 0, but it would require
+			// an additional currentMerger != nil check, since an ID will always come after
+			// an empty ID. Since checking currentMerger is already required when a lower ID
+			// is found, simply combine it with that check.
```

Comment:
I stared at this comment for many seconds, and I still don't understand what it means 😞 . Can you please elaborate?